src/LibraryManager.Vsix/Contracts/HostInteraction.cs

```diff
@@ -2,94 +2,121 @@
             }
 
-            Logger.Log(string.Format(LibraryManager.Resources.Text.FileWrittenToDisk, path.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
+            Logger.Log(string.Format(LibraryManager.Resources.Text.FileWrittenToDisk, relativePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
 
             return true;
         }
 
-        public void DeleteFiles(params string[] relativeFilePaths)
+        public async Task<bool> DeleteFilesAsync (IEnumerable<string> relativeFilePaths, CancellationToken cancellationToken)
         {
-            foreach (string relativeFilePath in relativeFilePaths)
+            cancellationToken.ThrowIfCancellationRequested();
+
+            List<string> absolutePaths = new List<string>();
+
+            foreach (string filePath in relativeFilePaths)
             {
-                string absoluteFile = new FileInfo(Path.Combine(WorkingDirectory, relativeFilePath)).FullName;
+                string absoluteFilePath = Path.Combine(WorkingDirectory, filePath);
+                FileInfo file = new FileInfo(absoluteFilePath);
 
-                try
+                if (file.Exists)
                 {
-                    ProjectItem item = VsHelpers.DTE.Solution.FindProjectItem(absoluteFile);
-                    Project project = item?.ContainingProject;
-
-                    if (project != null)
-                    {
-                        item.Delete();
-                    }
-                    else
-                    {
-                        if (File.Exists(absoluteFile))
-                        {
-                            VsHelpers.CheckFileOutOfSourceControl(absoluteFile);
-                            File.Delete(absoluteFile);
-                        }
-                    }
-
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleted, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
-                }
-                catch (Exception ex)
-                {
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleteFail, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
-                    Telemetry.TrackException("deletefilefailed", ex);
+                    absolutePaths.Add(absoluteFilePath);
                 }
             }
+
+            //Delete from project 
+            var logAction = new Action<string, LogLevel>((message, level) => { Logger.Log(message, level); });
+            Project project = VsHelpers.GetDTEProjectFromConfig(_configFilePath);
+            bool deleteFromProject = await VsHelpers.DeleteFilesFromProjectAsync(project, absolutePaths, logAction, cancellationToken);
+            if (deleteFromProject)
+            {
+                return true;
+            }
+
+            // Delete from file system 
```

Comment:
Is this the logic we want? If we fail to delete from project we delete from disk?

Reply:
In Core or WSP, this basically amounts to a retry since those are based on file-system (though WSP doesn't refresh).  For WAP the user would see the file go missing?  I think that might be ok, if it's just going to be restored later.  

On the other hand, I suppose on upgrades/uninstalls, they could get left with turds in their project file, but something triggered the delete...